x/ccv/provider/keeper/throttle.go

```diff
@@ -453,21 +472,28 @@ func (k Keeper) GetAllThrottledPacketData(ctx sdktypes.Context, consumerChainID
 		case slashPacketData:
 			d := ccvtypes.SlashPacketData{}
 			if err := d.Unmarshal(bz[1:]); err != nil {
-				panic(fmt.Sprintf("failed to unmarshal slash packet data: %v", err))
+				k.Logger(ctx).Error(fmt.Sprintf("failed to unmarshal slash packet data: %v", err))
+				continue
 			}
 			slashData = append(slashData, d)
 			rawOrderedData = append(rawOrderedData, d)
 		case vscMaturedPacketData:
 			d := ccvtypes.VSCMaturedPacketData{}
 			if err := d.Unmarshal(bz[1:]); err != nil {
-				panic(fmt.Sprintf("failed to unmarshal vsc matured packet data: %v", err))
+				k.Logger(ctx).Error(fmt.Sprintf("failed to unmarshal vsc matured packet data: %v", err))
+				continue
 			}
 			vscMaturedData = append(vscMaturedData, d)
 			rawOrderedData = append(rawOrderedData, d)
 		default:
-			panic(fmt.Sprintf("invalid packet data type: %v", bz[0]))
+			k.Logger(ctx).Error(fmt.Sprintf("invalid packet data type: %v", bz[0]))
+			continue
+		}
+		_, ibcSeqNum, err := providertypes.ParseThrottledPacketDataKey(iterator.Key())
+		if err != nil {
+			k.Logger(ctx).Error(fmt.Sprintf("failed to parse throttled packet data key: %v", err))
+			continue
```

Comment:
I'm not sure I understand the justification for this. Why not just panic?
If you can't unmarshal there's a serious problem so you should assume unexpected behavior in general, and panic. 
Failures tend to correlate so if there's one unmarshal failure there might be many more, which won't be handled, and things could deteriorate.

Reply:
See comment for this method

> // Since this method executes on query, no panics are explicitly included.

`GetAllThrottledPacketData` is only used by queries and tests, it's best not to expose panics to queries imo. For example, maybe there's a reason why this method would fail, while `GetSlashAndTrailingData` would not fail. The latter method should be held to a higher standard because it runs on endblock

For any code that runs on txs and end/begin blockers, I completely agree with your sentiments, granted that external data is properly validated somehow